hs_core/models.py

```diff
@@ -1381,6 +1381,20 @@ def get_crossref_deposit_xml(self, pretty_print=True):
 
         return '<?xml version="1.0" encoding="UTF-8"?>\n' + etree.tostring(ROOT, pretty_print=pretty_print)
 
+    @property
+    def verbose_name(self):
+        from hs_core.hydroshare.utils import get_resource_types
+        res_cls = None
+        for tp in get_resource_types():
+            if self.resource_type == tp.__name__:
+                res_cls = tp
+                break
+        if res_cls is not None:
+            res_obj = res_cls()
+            return res_obj._meta.verbose_name
+        else:
+            return self._meta.verbose_name
```

Comment:
@pkdash @alvacouch I just tested with `self.get_content_model()._meta.verbose_name`. It works correctly for us. I checked the issue we had in #1142 . The thing we tried there was `self._meta.verbose_name`. I stated that `self._meta.verbose_name` did not work correctly. It only returned 'Generic' to us. And I modified it to `self.get_content_model()._meta.verbose_name` in issue #1142 , but I did not clearly announce that `self.get_content_model()._meta.verbose_name` worked properly in the discussion. Then @pkdash came up with the current solution. I just change it back to `return self.get_content_model()._meta.verbose_name`. Sorry for the misleading.